src/rust/engine/fs/store/src/lib.rs

```diff
@@ -238,6 +238,53 @@ impl Store {
       .to_boxed()
   }
 
+  /// Store the given bytes buffer as a file at the given path
+  pub fn store_file_with_path(&self, path: PathBuf, contents: Bytes) -> BoxFuture<Digest, String> {
```

Comment:
I'd probably inline this at the call site - taking a file-like thing (path/bytes) and returning a `Digest`, it's unclear whether this returns a `Digest` of a file or a directory, whereas having the caller use something which returns a `Snapshot`, and then do the manipulation itself to project out the `Digest` is more clear.